doc/source/configuration/vault.rst

```diff
@@ -0,0 +1,207 @@
+      # 503: sealed (required for bootstrapping)
+      http-check expect rstatus (200|501|503)
+
+   {% for host in groups['control'] %}
+   {% set host_name = hostvars[host].ansible_facts.hostname %}
+   {% set host_ip = 'api' | kolla_address(host) %}
+      server {{ host_name }} {{ host_ip }}:8200 check check-ssl verify none inter 2000 rise 2 fall 5
+   {% endfor %}
+   {% endraw %}
+
+2. Deploy HAProxy with the new Vault service configuration:
+
+.. code-block::
+
+   kayobe overcloud service deploy -kt haproxy
+
+Setup Vault HA on the overcloud hosts
+-------------------------------------
+
+1. Run vault-deploy-overcloud.yml custom playbook
+
+.. code-block::
+
+   kayobe playbook run ansible/vault-deploy-overcloud.yml
+
+2. Encrypt overcloud vault keys (use proper location of vault password file)
+
+.. code-block::
+
+   ansible-vault encrypt --vault-password-file ~/vault.pass vault/overcloud-vault-keys.json
+
+Create the internal TLS certificates
+------------------------------------
+
+1. Run the playbook
+
+.. code-block::
+
+   kayobe playbook run ansible/vault-deploy-internal-tls.yml
+
+2. Use ansible-vault to encrypt the PEM bundle in kayobe-config/etc/kayobe/kolla/certificates/haproxy-internal.pem. Commit the PEM bundle and root CA to the kayobe configuration.
+
+.. code-block::
+
+   ansible-vault encrypt --vault-password-file ~/vault.pass kolla/certificates/haproxy-internal.pem
+
+Create the backend TLS certificates
+-----------------------------------
+
+1. Run the playbook
+
+.. code-block::
+
+   kayobe playbook run ansible/vault-deploy-backend-tls.yml
+
+2. Use ansible-vault to encrypt the keys in kayobe-config/etc/kayobe/kolla/certificates/<controller>-key.pem. Commit the certificates and keys to the kayobe configuration.
+
+.. code-block::
+
+   ansible-vault encrypt --vault-password-file ~/vault.pass kolla/certificates/<controller>-key.pem
+
+Enable the required TLS variables in kayobe and kolla
+-----------------------------------------------------
+
+1. Set the following in kayobe-config/etc/kayobe/kolla.yml or if environments are being used etc/kayobe/environments/$KAYOBE_ENVIRONMENT/kolla.yml
+
+.. code-block::
+
+   # Whether TLS is enabled for the internal API endpoints. Default is 'no'.
+   kolla_enable_tls_internal: yes
+
+2. Set the following in etc/kayobe/kolla/globals.yml or if environments are being used etc/kayobe/environments/$KAYOBE_ENVIRONMENT/kolla/globals.yml
+
+.. code-block::
+
+   # Internal TLS configuration
+   # Copy the self-signed CA into the kolla containers
+   kolla_copy_ca_into_containers: "yes"
+   # Use the following trust store within the container
+   openstack_cacert: "{{ '/etc/pki/tls/certs/ca-bundle.crt' if os_distribution in ["centos", "rocky"] else '/etc/ssl/certs/ca-certificates.crt' }}"
+
+   # Backend TLS config
+   # Enable backend TLS
+   kolla_enable_tls_backend: "yes"
+
+   # If using RabbitMQ TLS:
+   rabbitmq_enable_tls: "yes"
+
+3. Deploy backend and internal TLS
```

Comment:
What is the impact of this on an existing deployment?

Reply:
Impact is similar to the upgrade - restart of all services.

Reply:
It's a bit worse than upgrade, since endpoints become unreachable after HAProxy restart until the endpoint catalogue and client has been reconfigured to use HTTPS.

Reply:
indeed, yesterday I've tested it while enabling this model in UIBK. One needs to be patient and wait for all services to reconfigure as you said.

Reply:
Ok. Let's call that out here as there are some deployments that are sensitive to downtime.